sdk/clientcore/tools/annotation-processor/src/main/java/io/clientcore/tools/codegen/AnnotationProcessor.java

```diff
@@ -0,0 +1,196 @@
+                .map(roundEnv::getElementsAnnotatedWith)
+                .flatMap(Set::stream)
+                .filter(element -> element.getKind().isInterface())
+                .forEach(element -> {
+                    if (element.getAnnotation(ServiceInterface.class) != null) {
+                        this.processServiceInterface(element);
+                    }
+                });
+
+        return true;
+    }
+
+
+    private void processServiceInterface(Element serviceInterface) {
+        TemplateInput templateInput = new TemplateInput();
+
+        // work out some global details about this service interface
+        final String serviceInterfaceFQN = serviceInterface.asType().toString();
+
+        String packageName = null;
+        int lastDot = serviceInterfaceFQN.lastIndexOf('.');
+        if (lastDot > 0) {
+            packageName = serviceInterfaceFQN.substring(0, lastDot);
+        }
+
+        final String serviceInterfaceShortName = serviceInterfaceFQN.substring(lastDot + 1);
+        final String serviceInterfaceImplFQN = serviceInterfaceFQN + "Impl";
+        final String serviceInterfaceImplShortName = serviceInterfaceImplFQN.substring(lastDot + 1);
+
+        templateInput.setPackageName(packageName);
+        templateInput.setServiceInterfaceFQN(serviceInterfaceFQN);
+        templateInput.setServiceInterfaceShortName(serviceInterfaceShortName);
+        templateInput.setServiceInterfaceImplShortName(serviceInterfaceImplShortName);
+
+        // Read ServiceInterface.host() value from any enclosed interfaces
+        if (serviceInterface.getAnnotation(ServiceInterface.class) != null) {
+            templateInput.setHost(serviceInterface.getAnnotation(ServiceInterface.class).host());
+        }
+
+        // add all imports
+        addImports(templateInput);
+
+        // iterate through all methods in the interface, collecting all that are @HttpRequestInformation
+        // and then generating a method implementation for each one
+        templateInput.setHttpRequestContexts(serviceInterface.getEnclosedElements().stream()
+                .filter(element -> element.getKind() == ElementKind.METHOD)
+                .filter(element -> element.getAnnotation(HttpRequestInformation.class) != null)
+                .map(ExecutableElement.class::cast)
+                .map(e -> createHttpRequestContext(e, templateInput))
+                .collect(Collectors.toList()));
+
+        // template input set UnexpectedResponseExceptionDetails
+        templateInput.setUnexpectedResponseExceptionDetails(serviceInterface.getEnclosedElements().stream()
+                .filter(element -> element.getKind() == ElementKind.METHOD)
+                .filter(element -> element.getAnnotation(HttpRequestInformation.class) != null)
+                .map(ExecutableElement.class::cast)
+                .map(e -> e.getAnnotation(UnexpectedResponseExceptionDetail.class))
+                .collect(Collectors.toList()));
```

Comment:
Given the streams here have the same initial filters and mapping to `ExecutableElement` could we create an initial list and work from there?